### PR TITLE
support scala 2.13

### PR DESCRIPTION
**Pull request by WellingR**
Unfortunately scala-collection-compat has not yet been released for the 2.13.0-RC1. Therefore I did an attempt to upgrade to 2.13.0-M5.

This is my work so far. I have resoved all binary-compatibility issues/
I intend to upgrade to 2.13.0-RC1 as soon as scala-collection-compat has been released.

Todos
- [x] Upgrade to scala-collection-compat 1.0.0 as soon as it has been released
- [x] Upgrade to scala 2.13.0-RC1 (or later) as soon as scala-collection-compat has been released for this scala version
- [x] Fix travis issue: "The job exceeded the maximum time limit for jobs, and has been terminated."(perhaps we can build the different scala versions separately)
- [ ] Fix failing test on 2.13: com.typesafe.slick.testkit.tests.NestingTest.testNestedTuples

### CODE REVIEW

File: slick/src/main/scala/slick/dbio/DBIOAction.scala [hunk 1]
```diff
@@ -202,22 +202,22 @@ object DBIOAction {
...
     }
-    val grouped = groupBySynchronicity[R, E](in.asInstanceOf[TraversableOnce[DBIOAction[R, NoStream, E]]])
+    val grouped = groupBySynchronicity[R, E](in.asInstanceOf[IterableOnce[DBIOAction[R, NoStream, E]]])
     grouped.length match {
-      case 0 => DBIO.successful(cbf().result())
+      case 0 => DBIO.successful(cbf.newBuilder.result())
       case 1 => sequenceGroupAsM(grouped.head)
       case n =>
-        grouped.foldLeft(DBIO.successful(cbf(in)): DBIOAction[mutable.Builder[R, M[R]], NoStream, E]) { (ar, g) =>
+        grouped.foldLeft(DBIO.successful(cbf.newBuilder): DBIOAction[mutable.Builder[R, M[R]], NoStream, E]) { (ar, g) =>
```
Comment (WellingR) [inline slick/src/main/scala/slick/dbio/DBIOAction.scala]: I was not completely sure how to fix this. Passing the `in` parameter does not seem to be possible. The question is,  is this new implementation correct?

File: slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/InsertTest.scala [hunk 2]
```diff
@@ -196,11 +196,12 @@ class InsertTest extends AsyncTest[JdbcTestDB] {
     val ts = TableQuery[T]
 
     for {
-      _ <- ts.schema.create
-      _ <- ts ++= Seq((1, "a"), (2, "b"))
-      _ <- ts.insertOrUpdate((3, "c")).map(_ shouldBe 1)
-      _ <- ts.insertOrUpdate((1, "d")).map(_ shouldBe 1)
-      _ <- ts.sortBy(_.id).result.map(_ shouldBe Seq((1, "d"), (2, "b"), (3, "c")))
+      // FIXME using underscores here causes a compile error, is this a scala 2.13.0-RC1 bug?
```
Comment (WellingR) [inline slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/InsertTest.scala]: @szeiger Do you have any idea if this is perhaps a scala 2.13 bug? It seems that the old version of this code does not compile on 2.13.0-RC1

**Comment by pjfanning** [comment]
The issue tracking this is https://github.com/slick/slick/issues/1958

**Comment by WellingR** [comment]
Closing this in favour of https://github.com/slick/slick/pull/1966

**Pull request closed without merging**